sdk/identity/azure-identity/tests/test_managed_identity.py

```diff
@@ -489,3 +489,39 @@ def send(request, **_):
         token = ManagedIdentityCredential(transport=mock.Mock(send=send)).get_token(scope)
         assert token.token == access_token
         assert token.expires_on == expires_on
+
+
+def test_azure_arc():
+    """Azure Arc 2019-08-15"""
+
+    access_token = "****"
+    expires_on = 42
+    identity_endpoint = "http://localhost:42/token"
+    imds_endpoint = "http://localhost:42"
+    scope = "scope"
+
+    def send(request, **_):
```

Comment:
It ideally should, but I'm not sure how to add it to the test so I started with this since I can manually test the challenge protocol

Reply:
It would be neater to use the pytest [tmpdir](https://docs.pytest.org/en/latest/tmpdir.html) but I'm not familiar enough with it to rattle off code (do check it out if you have time) so here's a sketch of my first thought:
```py
expected_key = "topsecret"
open("keyfile", "wt").write(expected_key)

def send(request, **_):
    # assert things that should be true of all requests like url, query,
    # method, Metadata header
    if "Authorization" in request.headers:
        assert request.headers["Authorization"] = "Basic " + expected_key
        return mock_response(...)
    return mock_response(headers={"WWW-Authenticate": ...})

transport = mock.Mock(send=mock.Mock(wraps=send))
with mock.patch("os.environ", ...):
    credential = ManagedIdentityCredential(transport=transport)
try:
    token = credential.get_token(scope=scope)
except:
    raise
finally:
    # delete the file you wrote
```

Reply:
My second thought is better--the test needs to verify the credential sends two requests and successfully completes the challenge, which will be easier using `validating_transport` instead of a simple `send` method.